### PR TITLE
add vertex, edge, and fill color properties

### LINKED ISSUE #658: fill_color, edge_color, vertex_color `@properties` for selectors

### PR DESCRIPTION
**Pull request by clewis7**
closes #658

### CODE REVIEW

**Comment by clewis7** [comment]
Wow switching back to Python from C is weird :anguished:

**Comment by clewis7** [comment]
https://github.com/user-attachments/assets/18d4c3db-cdaf-4acc-83e4-1e4c8c0dc8c6

File: fastplotlib/graphics/selectors/_base_selector.py [hunk 1]
```diff
@@ -40,6 +40,40 @@ class BaseSelector(Graphic):
...
+    @property
+    def fill_color(self) -> str | Sequence[float] | None:
+        return self._fill_color
+
+    @fill_color.setter
+    def fill_color(self, color: str | Sequence[float]):
+        for fill in self._fill:
+            fill.material.color = color
+            self._original_colors[fill] = color
+        self._fill_color = color
```
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_base_selector.py:52]: Convert into a pygfx.Color and store that

File: fastplotlib/graphics/selectors/_base_selector.py [hunk 2]
```diff
@@ -40,6 +40,40 @@ class BaseSelector(Graphic):
     def axis(self) -> str:
         return self._axis
 
+    @property
+    def fill_color(self) -> str | Sequence[float] | None:
```
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_base_selector.py:44]: These should always just return pygfx.Color

File: fastplotlib/graphics/selectors/_linear.py [hunk 3]
```diff
@@ -99,6 +99,9 @@ def __init__(
             name of linear selector
 
         """
+        self._fill_color = None
```
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_linear.py:102]: Set the initial values from the user args?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: fastplotlib/graphics/selectors/_linear.py (hunk 3 above)
Comment (clewis7) [reply fastplotlib/graphics/selectors/_linear.py:102]: Linear selector is weird: it uses an inner and outer line so it doesn't conform to the vertices, edges, and fill schema...

Comment (kushalkolar) [reply fastplotlib/graphics/selectors/_linear.py:102]: only the inner line color should be set, rename the kwargs `edge_color`?

Comment (clewis7) [reply fastplotlib/graphics/selectors/_linear.py:102]: yeah, I think it makes it more uniform

File: fastplotlib/graphics/selectors/_base_selector.py [hunk 4]
```diff
@@ -41,34 +42,37 @@ def axis(self) -> str:
...
 
     @property
-    def edge_color(self) -> str | Sequence[float] | None:
+    def edge_color(self) -> pygfx.Color:
         return self._edge_color
 
     @edge_color.setter
     def edge_color(self, color: str | Sequence[float]):
+        color = pygfx.Color(color)
         for edge in self._edges:
```
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_base_selector.py:76]: so this is going to change the outer and inner line for `LinearSelector`, override the `@property` in linear selector?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: fastplotlib/graphics/selectors/_base_selector.py
Comment (kushalkolar) [inline fastplotlib/graphics/selectors/_base_selector.py:1]: I forgot to mention docstrings for all the properties :smile:

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Review by kushalkolar — APPROVED** [review]

**Pull request merged**